plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListener.java

```diff
@@ -0,0 +1,420 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineage.ColumnLineageDatasetFacetBuilder;
+import io.openlineage.client.OpenLineage.DatasetFacetsBuilder;
+import io.openlineage.client.OpenLineage.InputDataset;
+import io.openlineage.client.OpenLineage.InputDatasetBuilder;
+import io.openlineage.client.OpenLineage.JobBuilder;
+import io.openlineage.client.OpenLineage.OutputDataset;
+import io.openlineage.client.OpenLineage.RunEvent;
+import io.openlineage.client.OpenLineage.RunFacet;
+import io.openlineage.client.OpenLineage.RunFacetsBuilder;
+import io.openlineage.client.OpenLineageClient;
+import io.trino.plugin.openlineage.config.OpenLineageListenerConfig;
+import io.trino.spi.eventlistener.BaseViewReferenceInfo;
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.eventlistener.FilterMaskReferenceInfo;
+import io.trino.spi.eventlistener.OutputColumnMetadata;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryContext;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import io.trino.spi.eventlistener.QueryFailureInfo;
+import io.trino.spi.eventlistener.QueryIOMetadata;
+import io.trino.spi.eventlistener.QueryMetadata;
+import io.trino.spi.eventlistener.QueryOutputMetadata;
+import io.trino.spi.eventlistener.QueryStatistics;
+import io.trino.spi.eventlistener.TableInfo;
+import io.trino.spi.resourcegroups.QueryType;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class OpenLineageListener
+        implements EventListener
+{
+    private static final Logger logger = Logger.get(OpenLineageListener.class);
+
+    private final OpenLineage openLineage = new OpenLineage(URI.create("https://github.com/trinodb/trino/plugin/trino-openlineage"));
+    private final OpenLineageClient client;
+    private final String namespace;
+    private final Set<QueryType> includeQueryTypes;
+    private final ObjectMapper queryStatisticsMapper;
+
+    @Inject
+    public OpenLineageListener(OpenLineageClient client, OpenLineageListenerConfig listenerConfig)
+            throws Exception
+    {
+        this.client = requireNonNull(client);
```

Comment:
Both Spark and Flink open lineage integration has made it configurable to name the `job.namespace` if an organization wants a custom name. Although there is [dataset naming conventions](https://openlineage.io/docs/spec/naming), there is no naming convention followed for `job.namespace` for different integrations.


I recently asked in the OL community channel as well
https://openlineage.slack.com/archives/C01CK9T7HKR/p1713900040070319

[Flink](https://github.com/OpenLineage/OpenLineage/blob/a97ae0bf33859855c33212d37735a2f520794eb4/integration/flink/app/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java#L55) override key:  `execution.job-listener.openlineage.namespace`

[Spark](https://github.com/OpenLineage/OpenLineage/blob/a97ae0bf33859855c33212d37735a2f520794eb4/integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java#L39) override key: `spark.openlineage.namespace`

We should also give a similar override capability for Trino with `openlineage-event-listener.trino.namespace` property and can keep the default value as `trino://{host}:port`

Reply:
true, added `openlineage-event-listener.namespace` for overriding job namespace